src/main/java/net/datafaker/providers/base/Currency.java

```diff
@@ -2,18 +2,25 @@
 
 /**
  * @since 0.8.0
+ * @deprecated since 2.3.0. For removal in 3.0.0 version. This faker is deprecated due to the existence
+ * of duplicate {@link Money} faker with similar methods. Use {@link Money} instead.
  */
+@Deprecated(since = "2.3.0", forRemoval = true)
 public class Currency extends AbstractProvider<BaseProviders> {
 
     public Currency(BaseProviders faker) {
         super(faker);
     }
 
+    /**
+     * @deprecated since 2.3.0. For removal in 3.0.0 version. Use {@link Money#currency} instead.
```

Comment:
IMHO if you're going to javadoc this then also include:
`@Deprecated(since = "2.2.0", forRemoval = true)`

Same throughout.